### PR TITLE
add go example

### PR DESCRIPTION
**Pull request by Yakiyo**
## What does this Pull Request change?
This pr adds an unofficial example in Go
## I agree that my Pull Request:

- Follows the [Contribution Guidelines](https://github.com/nekos-best/docs#contribution-guidelines)
- Follows [GitHub's Community Guidelines](https://docs.github.com/en/github/site-policy/github-community-guidelines)
- Contains an example that is well documented and is ordered alphabetically in [SUMMARY.md](https://github.com/nekos-best/docs/blob/main/src/SUMMARY.md)
- Contains an example that is not a duplicate of an already existing example.

## Any additional notes?

None

### CODE REVIEW

**Comment by Thusuzzee** [comment]
wtf is Go? I only get my coffee to go

**Comment by oriel-beck** [comment]
It's Rust's friend

File: src/examples/unofficial/go.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+fmt.Println(res[0].Url)
+```
+
+### Fetch a file
+```go
+import "os"
+
+res, _ := nb.FetchFile("pat")
+
+os.WriteFile("image.png", res.Data, 0644)
```
Comment (oriel-beck) [inline src/examples/unofficial/go.md:44]: The API no longer supports numbered IDs, please change your example to use a UUID.

**Review by oriel-beck — DISMISSED** [review]
Fix your code as well if you expect a number for the image ID instead of a UUID.

**Comment by thund-x** [comment]
> It's Rust's friend

Don't disrespect Rust like that

**Pull request merged**